runtime/db/modusdb.go

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2024 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package db
+
+import (
+	"context"
+
+	"github.com/hypermodeinc/modus/runtime/logger"
+	"github.com/hypermodeinc/modusdb"
+)
+
+var dataDir = "data"
+var GlobalModusDbEngine *modusdb.Engine
+
+func InitModusDb(ctx context.Context) {
+	// Initialize the database connection.
+	var err error
+	GlobalModusDbEngine, err = modusdb.NewEngine(modusdb.NewDefaultConfig(dataDir))
+	if err != nil {
+		logger.Fatal(ctx).Err(err).Msg("Failed to initialize modusdb.")
+		return
+	}
+}
```

Comment:
We should only be initializing Modus DB in development, and we should use the Modus home directory for the root of the storage.  Consider:

```suggestion
func InitModusDb(ctx context.Context) {
	if config.IsDevEnvironment() {
		dataDir := filepath.Join(app.ModusHomeDir(), "data")
		if eng, err := modusdb.NewEngine(modusdb.NewDefaultConfig(dataDir)); err != nil {
			logger.Fatal(ctx).Err(err).Msg("Failed to initialize modusdb.")
		} else {
			GlobalModusDbEngine = eng
		}
	}
}
```

You can add the following to `/runtime/app/app.go`:
```
func ModusHomeDir() string {
	modusHome := os.Getenv("MODUS_HOME")
	if modusHome != "" {
		userHome, _ := os.UserHomeDir()
		modusHome = filepath.Join(userHome, ".modus")
	}
	return modusHome
}
```